packages/core/src/util.ts

```diff
@@ -176,48 +176,53 @@ export class Util {
     static escapeRegExp(str: string): string {
         return str.replace(/[-/\\^$*+?.()|[\]{}]/g, '\\$&'); // $& means the whole matched string
     }
-
-    static findImpactedTests(
-        testsDepsMap: TestsDependenciesMap | null,
+    static async findImpactedTests(
+        testsDepsMap: TestsDependenciesMap | null, 
         tests: Test[],
         changedFilesSet: Set<string>
-    ): ID[] {
-        const impactedTests = new Set<ID>();
-        // skip if not diff exists or testFiles do not have testDeps
-        if (changedFilesSet.size === 0 || testsDepsMap === null) {
-            return [];
+    ): Promise<[ID[], boolean]>  {
+        // 
+        let executeAllTests = false;
+        // skip listing of dependencies if no changed files or smart mode not available
+        if (changedFilesSet.size === 0 || !await this.isSmartSelectAvailable()) {
```

Comment:
instead of `isSmartSelectAvailable` can we not check the null value of `testsDepsMap`? Or is it that `testsDepsMap` can be null in other cases too?

Reply:
Its a kind of rare case, but in case a file has no internal dependencies then it can be null

```
import assert from 'assert';

describe('Some simple test', () => {
  it('should return true', () => {
    assert.strictEqual(true, true);
  });
});```